libraries/AP_OSD/AP_OSD_MSP_DisplayPort.cpp

```diff
@@ -83,7 +86,47 @@ void AP_OSD_MSP_DisplayPort::clear(void)
 
 void AP_OSD_MSP_DisplayPort::write(uint8_t x, uint8_t y, const char* text)
 {
-    _displayport->msp_displayport_write_string(x, y, 0, text);
+#if AP_MSP_INAV_FONTS_ENABLED
+    const AP_MSP *msp = AP::msp();
+    if (msp && msp->is_option_enabled(AP_MSP::Option::DISPLAYPORT_INAV_SYMBOLS)) {
+        uint8_t idx = 0;
+        uint8_t buf_idx = 0;
+        uint8_t last_inav_table = 0xFF; // 0xFF == unknown
+        uint8_t x_offset = 0;
+        const uint8_t max_idx = MIN((uint8_t)(strlen(text)-1), DISPLAYPORT_WRITE_BUFFER_MAX_LEN-2);
+        uint8_t inav_symbol;
+        uint8_t inav_table;
+        while (text[idx] != 0 && idx <= max_idx)
+        {
+            // transcode
+            inav_symbol = ap_to_inav_symbols_map[(uint8_t)text[idx]][0];
+            inav_table = ap_to_inav_symbols_map[(uint8_t)text[idx]][1];
+            
+            if (last_inav_table == 0xFF)
```

Comment:
should have {} around line, following code style

Reply:
done